homeassistant/components/samsungtv/bridge.py

```diff
@@ -737,9 +763,9 @@ async def _async_get_remote_under_lock(
 
     async def async_power_off(self) -> None:
         """Send power off command to remote."""
-        raise HomeAssistantError(
-            "Sending commands to encrypted TVs is not yet supported"
-        )
+        await self._async_send_commands([SendEncryptedRemoteKey.click("KEY_POWER")])
```

Comment:
Maybe we should send both?
```suggestion
        # some models use KEY_POWEROFF (eg. UE50H6400)
        await self._async_send_commands([SendEncryptedRemoteKey.click("KEY_POWEROFF")])
        # some models use KEY_POWER (eg. UN75JU641D)
        await asyncio.sleep(1)
        await self._async_send_commands([SendEncryptedRemoteKey.click("KEY_POWEROFF")])
```

Reply:
Does KEY_POWEROFF work on any encrypted models?

Reply:
If its confirmed to work on `UE50H6400` then probably sending both makes sense

Reply:
Or maybe if we know its an `H` send `KEY_POWEROFF` and a `J` send `KEY_POWER`

Reply:
If we want to support more models, I would say send both.
If we want to be strict, then maybe default to `KEY_POWEROFF` and use a known model list to override with `POWER_KEY`

Reply:
If you don't know for sure, send both.  If you do know for sure which one it takes, send that one

Reply:
How about this?
```suggestion
        power_off_commands: list[SendEncryptedRemoteKey] = []
        if self._model in self._MODEL_USES_POWER_OFF:
            power_off_commands.append(SendEncryptedRemoteKey.click("KEY_POWEROFF"))
        elif self._model in self._MODEL_USES_POWER:
            power_off_commands.append(SendEncryptedRemoteKey.click("KEY_POWER"))
        else:
            LOGGER.info(
                "Unknown power_off command for %s (%s): sending KEY_POWEROFF and KEY_POWER",
                self.host,
                self._model,
            )
            power_off_commands.append(SendEncryptedRemoteKey.click("KEY_POWEROFF"))
            power_off_commands.append(SendEncryptedRemoteKey.click("KEY_POWER"))
        await self._async_send_commands(power_off_commands)

```

Reply:
In my case, KEY_POWEROFF always works except when the YouTube app is open. In that case KEY_POWEROFF acts as a RETURN command, but the TV doesn't turn off. I guess that's a misunderstanding of the YouTube app.

Sending both commands to turn off the TV may be the best and easy solution, but I would also enable the 'on_turn_off' property so that, if the user defines it, a more reliable method of turning it on and off is enabled, such as using a relay, IR, or HDMI-CEC

Reply:
For now, I have purposefully ignored `on_turn_on` / `on_turn_off` properties, at least until we decide what we do about the old `on_turn_on` method. See https://github.com/home-assistant/core/pull/68185

Reply:
I have pushed up my changes for the double-key `KEY_POWER`/`KEY_POWEROFF`.
I have used `frame.report` to log it when unknown, so it should only get logged once and encourage users to add their model.

Reply:
@ooohaaa it would be interesting to have your results on model `UE48JU6400`.
Does it work with `KEY_POWER`? or with `KEY_POWEROFF`? or neither?
Does the token still work after power off, or does it only work on first attempt?

Reply:
@epenet I will be able to test this weekend. Starting with next monday I won't be available for two weeks.

> Does it work with `KEY_POWER`? or with `KEY_POWEROFF`? or neither?

How can I test this?
Or shall I just run the new custom componet like described below?

> The upstream library has been updated, so it should now all work correctly inside Home Assistant as a custom component.
> 
> I have pushed commit [2a59620](https://github.com/epenet/home-assistant-core/commit/2a596204284a666c43038fe712c7a5a71b37e38e) if anyone wants to have a go: https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/epenet/home-assistant-core/tree/2a596204284a666c43038fe712c7a5a71b37e38e/homeassistant/components/samsungtv
> 
> @ooohaaa you should be able to test it as well - no need for Python knowledge hopefully.

Reply:
You can use the new commit: a4ab29647f653142613e8cc52430a42f749ecb5e:
https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/epenet/home-assistant-core/tree/a4ab29647f653142613e8cc52430a42f749ecb5e/homeassistant/components/samsungtv

By default, it will send both commands and should report a warning in the logs.
Then to try just one command you can adjust lines 615-616 and see which one works for you:
https://github.com/epenet/home-assistant-core/blob/a4ab29647f653142613e8cc52430a42f749ecb5e/homeassistant/components/samsungtv/bridge.py#L615-L616

Reply:
@epenet i'm agreed with your solution, but there is an important thing you must know. The Samsung model identifier is composed of several parts:

- First two characters means region/continent (UE is from Europe)
- Following two characters means screen size (50 means 50")
- H6400 is the real model, regardless of screen size or region

I mean, an UN45H6400 model will work the same way as UE50H6400.

Reply:
😆 yes I already adjusted that in https://github.com/epenet/home-assistant-core/pull/71/commits/63e96ec6d98ef8b2991608eba6005b9d05f0d642